### ISSUE TITLE
Example LSTM-network for time series binary classification

### ISSUE THREAD

**Issue by GeorgyKonoplich** [facebookarchive/caffe2#849]
Hello everyone ! 
I just wanted to know how to create model for time series binary classification . I try to change  [char_rnn.py example](https://github.com/caffe2/caffe2/blob/master/caffe2/python/examples/char_rnn.py). There is a code:

        model = model_helper.ModelHelper(name="rnn")

        input_blob, seq_lengths, hidden_init, cell_init, target = \
            model.net.AddExternalInputs(
                'input_blob',
                'seq_lengths',
                'hidden_init',
                'cell_init',
                'target',
            )

        hidden_output_all, self.hidden_output_last, _, self.cell_state = LSTM(
            model, input_blob, seq_lengths, (hidden_init, cell_init),
            3, self.hidden_size, scope="LSTM")
        output = brew.fc(
            model,
            self.hidden_output_last,
            None,
            dim_in=self.hidden_size,
            dim_out=1,
            axis=2
        )

        pred = model.net.Sigmoid(output, 'pred')
        loss = model.net.SigmoidCrossEntropyWithLogits([pred, target], 'loss')
        model.AddGradientOperators([loss])
        build_sgd(
            model,
            base_learning_rate=0.1,
            policy="step",
            stepsize=1,
            gamma=0.9999
        )
But I get an Exception: No gradient registered for RecurrentNetwork. Exception from creating the gradient op: [enforce fail at operator_gradient.h:137] g_output_.at(i).IsDense(). Gradient of output LSTM/hidden_t_all is either sparse or not provided.

Any thoughts about how to create model?
Thanks!

**Comment by akyrola** [comment]
I think problem is that the LSTM only supports gradient over the hidden_output_all blob. Instead you, are passing the last hidden output to FC, and to the loss.
